### ISSUE TITLE
Build: OpenBSD 7.1 builtins.sh failing for cd -P tests.

### ISSUE THREAD

**Issue by hyenias** [ksh93/ksh#480]
I upgraded from OpenBSD 7.0 to 7.1. Same builtins.sh tests are now failing on OpenBSD 7.1 as of https://github.com/ksh93/ksh/commit/490edfe28e5c4ee45f8542e6fedc3afc7e03f971 that were fixed back in #229.
```
openbsd$ uname -a
OpenBSD openbsd.my.domain 7.1 GENERIC.MP#3 amd64
openbsd$ bin/shtests
#### Regression-testing /mnt/tmp/ksh/arch/openbsd.amd64-64/bin/ksh ####
...
test builtins begins at 2022-05-21+19:28:34
/mnt/tmp/ksh/src/cmd/ksh93/tests/builtins.sh[1499]: cd: .: [No such file or directory]
	builtins.sh[1501]: FAIL: cd -P without -e exits with error status if $PWD doesn't exist (expected 0, got 1)
/mnt/tmp/ksh/src/cmd/ksh93/tests/builtins.sh[1504]: cd: .: [No such file or directory]
	builtins.sh[1506]: FAIL: cd -eP doesn't fail if $PWD doesn't exist (expected 1, got 2)
test builtins failed at 2022-05-21+19:28:44 with exit code 2 [ 287 tests 2 errors ]
test builtins(C.UTF-8) begins at 2022-05-21+19:28:44
/mnt/tmp/ksh/src/cmd/ksh93/tests/builtins.sh[1499]: cd: .: [No such file or directory]
	builtins.sh[1501]: FAIL: cd -P without -e exits with error status if $PWD doesn't exist (expected 0, got 1)
/mnt/tmp/ksh/src/cmd/ksh93/tests/builtins.sh[1504]: cd: .: [No such file or directory]
	builtins.sh[1506]: FAIL: cd -eP doesn't fail if $PWD doesn't exist (expected 1, got 2)
test builtins(C.UTF-8) failed at 2022-05-21+19:28:54 with exit code 2 [ 287 tests 2 errors ]
test builtins(shcomp) begins at 2022-05-21+19:28:54
/tmp/ksh93.shtests.44923.13178/shcomp-builtins.ksh[1499]: cd: .: [No such file or directory]
	shcomp-builtins.ksh[1501]: FAIL: cd -P without -e exits with error status if $PWD doesn't exist (expected 0, got 1)
/tmp/ksh93.shtests.44923.13178/shcomp-builtins.ksh[1504]: cd: .: [No such file or directory]
	shcomp-builtins.ksh[1506]: FAIL: cd -eP doesn't fail if $PWD doesn't exist (expected 1, got 2)
test builtins(shcomp) failed at 2022-05-21+19:29:04 with exit code 2 [ 287 tests 2 errors ]
```

**Comment by McDutchie** [comment]
I was already getting these failures on QNX and Solaris.

Can you confirm that commit e5e19cdf27ac31ce49f2afe851697f9366bdf347 fixed them for OpenBSD 7.1 as well?

**Comment by hyenias** [comment]
Yes, confirmed. All fixed now. Thank you.
